includes/class-wc-gateway-ppec-cart-handler.php

```diff
@@ -182,6 +183,45 @@ protected function start_checkout() {
 		}
 	}
 
+	/**
+	 * Store checkout form data in customer session.
+	 *
+	 * @since 1.6.4
+	 */
+	protected function set_customer_data( $data ) {
+		$customer = WC()->customer;
+
+		$shipping_prefix = isset( $data['ship_to_different_address'] ) ? 'shipping' : 'billing';
+
+		$customer->set_shipping_address( $data[ $shipping_prefix . '_address_1' ] );
```

Comment:
Let's avoid PHP Warnings by checking for these keys before accessing, and also sanitize the user input.

Using core WC as an example ([#](https://github.com/woocommerce/woocommerce/blob/fd722da88ab5c42afe55965773229e33aec74ca8/includes/class-wc-form-handler.php#L206)):

```php
$shipping_address = empty( $data[ $shipping_prefix . '_address_1' ] ) ? '' : wc_clean( $data[ $shipping_prefix . '_address_1' ] );
$customer->set_shipping_address( $shipping_address );
```

Reply:
👍 Done in f52e141